manticore/core/smtlib/expression.py

```diff
@@ -690,19 +687,18 @@ def cast(self, possible_array):
             return arr
         raise ValueError  # cast not implemented
 
-    def cast_index(self, index):
+    def cast_index(self, index: int) -> Union[int, "BitVecConstant"]:
```

Comment:
Removed assert below also allows for the possibility of index being a BitVec.
```suggestion
    def cast_index(self, index: Union[int, "BitVec"]) -> Union[int, "BitVecConstant"]:
```

Reply:
* Ah! I missed that. Thanks for catching! I think the one other below also has to change. Will make the changes and ensure once again.
* Also, I was trying to figure out the annotations to `Any`, could not find a good way that works uniformly for all instances yet.

Reply:
Ah, sorry I wasn't more clear about this, but for `*args` and `**kwargs`, I'm fine with removing the `: Any` altogether (provided it passes `mypy`)

Reply:
I thought about that too, thanks for the clarification!